feat: adiciona modal para edição de hábitos

ThaisMap opened the pull request:
- substitui as páginas de edição e criação de hábitos por um modal

Closes JOR-61

ThaisMap committed:
feat: adiciona modal para edição de hábitos

- substitui a página de edição e criação de hábitos por um modal

ThaisMap committed:
refactor: ajusta props do habitoCheckboxGroup

- tira a data das props do habitoCheckboxGroup
- tira o capitalize do título do modal de edição
- o eslint estava dando um aviso que o 'always-multiline' não era um valor válido

kellynvd reviewed (changes requested):
@ThaisMap encontrei os seguintes problemas testando:

- [x] Eu consigo criar e editar um hábito com os campos nome e emoji vazios. Acredito que devemos deixar ambos os campos obrigatórios
- [x]  Eu consigo adicionar letras e números no campo Emoji, deveria ser só emojis
- [x] Eu marquei um hábito > cliquei em editar  > editei o nome e salvei  a edição >  ao fazer isso a marcação foi removida então marquei novamente e salvei o registro. E esse registro o mesmo hábito duas vezes:

https://user-images.githubusercontent.com/13313889/132249407-7c53866f-17a1-47f0-a960-ac4ab0b9f758.mp4

ThaisMap committed:
fix: ajusta validações

- valida se o emoji e o nome estão preenchidos
- valida se texto no input de emoji é um emoji mesmo
- valida pelo id se o hábito está marcado ao invés do nome

ThaisMap committed:
fix: desmarcar hábito editado e mensagem de erro

- Não permitia desmarcar hábito depois de editá-lo
- Não exibia mensagem de erro ao tentar criar hábito sem preencher os campos
